core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -956,24 +976,24 @@ class SparkContext(config: SparkConf) extends Logging {
   }
 
   /**
-   * Get an RDD for a Hadoop-readable dataset from a Hadoop JobConf given its InputFormat and other
-   * necessary info (e.g. file name for a filesystem-based dataset, table name for HyperTable),
-   * using the older MapReduce API (`org.apache.hadoop.mapred`).
+   * Get an RDD for a Hadoop-readable dataset from a Hadoop `JobConf` given its `InputFormat`
+   * and other necessary info (e.g. file name for a filesystem-based dataset, table name
+   * for HyperTable), using the older MapReduce API (`org.apache.hadoop.mapred`).
    *
-   * @param conf JobConf for setting up the dataset. Note: This will be put into a Broadcast.
+   * @note Because Hadoop's `RecordReader` class re-uses the same Writable object for each
+   * record, directly caching the returned RDD or directly passing it to an aggregation
+   * or shuffle operation will create many references to the same object.
+   * If you plan to directly cache, sort, or aggregate Hadoop writable objects, you
+   * should first copy them using a `map` function.
+   * @param conf `JobConf` for setting up the dataset. Note: This will be put into a Broadcast.
    *             Therefore if you plan to reuse this conf to create multiple RDDs, you need to make
    *             sure you won't modify the conf. A safe approach is always creating a new conf for
    *             a new RDD.
    * @param inputFormatClass Class of the InputFormat
    * @param keyClass Class of the keys
    * @param valueClass Class of the values
-   * @param minPartitions Minimum number of Hadoop Splits to generate.
-   *
```

Comment:
I think it won't affect where it actually renders in the docs, and so yes I think it should be left where it is